common/lib/xmodule/xmodule/modulestore/__init__.py

```diff
@@ -775,6 +775,10 @@ def _is_set_on(key):
 
         for key, criteria in qualifiers.iteritems():
             is_set, value = _is_set_on(key)
+
+            if isinstance(criteria, dict) and '$exists' in criteria and criteria['$exists'] == is_set:
+                return True
```

Comment:
This seems broken. It should `continue` instead of `return True` in order to respect multiple criteria.